src/pkg/packager/compose.go

```diff
@@ -146,16 +146,25 @@ func (p *Packager) getChildComponent(parent types.ZarfComponent, pathAncestry st
 	// If it's OCI, we need to unpack the component tarball
 	if parent.Import.URL != "" {
 		dir := filepath.Join(cachePath, "components", child.Name)
+		componentTarball := fmt.Sprintf("%s.tar", dir)
 		parent.Import.Path = filepath.Join(parent.Import.Path, "components", child.Name)
-		if !utils.InvalidPath(dir) {
-			err = os.RemoveAll(dir)
+		if !utils.InvalidPath(componentTarball) {
+			if !utils.InvalidPath(dir) {
+				err = os.RemoveAll(dir)
+				if err != nil {
+					return child, fmt.Errorf("unable to remove composed component cache path %s: %w", cachePath, err)
+				}
+			}
+			err = archiver.Unarchive(componentTarball, filepath.Join(cachePath, "components"))
 			if err != nil {
-				return child, fmt.Errorf("unable to remove composed component cache path %s: %w", cachePath, err)
+				return child, fmt.Errorf("unable to unpack composed component tarball: %w", err)
+			}
+		} else {
+			// If the tarball doesn't exist (skeleton component had no local resources), we need to create the directory anyways in case there are actions
+			err := utils.CreateDirectory(dir, 0600)
```

Comment:
```suggestion
			err := utils.CreateDirectory(dir, 0700)
```
Directories should be `0700` - "execute" permissions for directories allows the user to enter the directory so many commands will fail trying to access a directory at `0600`